Dockerfile

```diff
@@ -17,37 +17,79 @@ ARG PG_MAJOR=13
+## container is something we'd like to avoid, we've seen multiple glibc
+## related bugs in our lifetime, adding multiple glibc versions in the mix
+## would make debugging harder.
+
+## Debian (and rust:debian) has served us well in the past, however even Debian's
+## latest release (bullseye, August 2021) cannot give us glibc 2.33.
+## Ubuntu however does give us glibc 2.33 - as Ubuntu is based upon Debian
+## the changes required are not that big for this Docker Image. Most of the
+## tools we use will be the same across the board, as most of our tools our
+## installed using external repositories.
+FROM ubuntu:21.04 AS compiler
 
 # We need full control over the running user, including the UID, therefore we
 # create the postgres user as the first thing on our list
 RUN adduser --home /home/postgres --uid 1000 --disabled-password --gecos "" postgres
 
-# CI/CD may benefit a lot from using a specific package mirror
-ARG DEBIAN_REPO_MIRROR=""
-RUN echo 'APT::Install-Recommends "0";\nAPT::Install-Suggests "0";' > /etc/apt/apt.conf.d/01norecommend \
-    && if [ "${DEBIAN_REPO_MIRROR}" != "" ]; then \
-        sed -i "s{http://.*.debian.org{http://${DEBIAN_REPO_MIRROR}{g" /etc/apt/sources.list; \
-    fi
+RUN echo 'APT::Install-Recommends "0";\nAPT::Install-Suggests "0";' > /etc/apt/apt.conf.d/01norecommend
 
 # Install the highlest level dependencies
 RUN apt-get update \
-    && apt-get install -y curl ca-certificates locales gnupg1 lsb-release
+    && apt-get install -y curl ca-certificates gpg-agent wget gpg locales gnupg1 lsb-release
+
+RUN apt-get purge libappstream4
+
+RUN mkdir -p /build
+RUN chmod 777 /build
+WORKDIR /build/
 
+RUN wget -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | gpg --dearmor > /usr/share/keyrings/postgresql.keyring
 RUN for t in deb deb-src; do \
-        echo "$t http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -s -c)-pgdg main" >> /etc/apt/sources.list.d/pgdg.list; \
+        echo "$t [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/postgresql.keyring] http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -s -c)-pgdg main" >> /etc/apt/sources.list.d/pgdg.list; \
     done
-RUN curl -s -o - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+
+RUN apt-get clean
+RUN apt-get update
 
 # The following tools are required for some of the processes we (TimescaleDB) regularly
 # run inside the containers that use this Docker Image
-RUN apt-get update && apt-get install -y less jq strace procps
+RUN apt-get install -y less jq strace procps
 
 # For debugging it is very useful if the Docker Image contains gdb(server). Even though it is
 # not expected to be running gdb in a live instance often, it simplifies getting backtraces from
 # containers using this image
 RUN apt-get install -y gdb gdbserver
 
+# The next 2 run instructions lines are directly copied from https://github.com/rust-lang/docker-rust/blob/master/1.54.0/bullseye/Dockerfile
```

Comment:
github links are best done with a commit hash instead of a branch, otherwise they break. And as I went to get a stable link, it seems this master link is broken already. (you can press `y` on a github page to turn a branch URL into a hash URL)

https://github.com/rust-lang/docker-rust/blob/21171fdd92e29acb045a41cd58b0d30d66aeaa7f/1.54.0/bullseye/Dockerfile is the stable link